### PR TITLE
fix/ff 3389

### PR DESCRIPTION
**Pull request by sameerank**
🎟️ fixes ff-3389

`new DateTime()` sets the instance to January 1, 0001 at 00:00:00.000
`DateTime.UtcNow` sets the instance to the current date and time in UTC

Also, added a test to confirm

### CODE REVIEW

File: dot-net-sdk/dto/AssignmentLogData.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ IReadOnlyDictionary<string, string> extraLoggging
         this.FeatureFlag = featureFlag;
         this.Allocation = allocation;
         this.Variation = variation;
-        this.Timestamp = new DateTime();
```
Comment (leoromanovsky) [inline dot-net-sdk/dto/AssignmentLogData.cs:33]: What terrible default behavior 😒

File: eppo-sdk-test/helpers/AssignmentLogDataTest.cs [hunk 2]
```diff
@@ -4,17 +4,30 @@ namespace eppo_sdk_test.helpers;
...
+    [Test]
+    public void ShouldBuildExperimentKey()
+    {
         Assert.That(assignmentLogData.Experiment, Is.EqualTo("feature-flag-allocation"));
     }
+
+    [Test]
+    public void ShouldSetAssignmentTimestamp()
+    {
+        Assert.That(assignmentLogData.Timestamp, Is.EqualTo(DateTime.UtcNow).Within(1).Milliseconds);
```
Comment (leoromanovsky) [inline eppo-sdk-test/helpers/AssignmentLogDataTest.cs:31]: Slick!

**Review by leoromanovsky — APPROVED** [review]

File: dot-net-sdk/dto/AssignmentLogData.cs (hunk 1 above)
Comment (aarsilv) [reply dot-net-sdk/dto/AssignmentLogData.cs:33]: 🤮

File: eppo-sdk-test/helpers/AssignmentLogDataTest.cs [hunk 3]
```diff
@@ -4,17 +4,30 @@ namespace eppo_sdk_test.helpers;
...
+    }
+
+    [Test]
+    public void ShouldBuildExperimentKey()
+    {
         Assert.That(assignmentLogData.Experiment, Is.EqualTo("feature-flag-allocation"));
     }
+
+    [Test]
+    public void ShouldSetAssignmentTimestamp()
```
Comment (aarsilv) [inline eppo-sdk-test/helpers/AssignmentLogDataTest.cs:29]: 🙌

File: eppo-sdk-test/helpers/AssignmentLogDataTest.cs (hunk 2 above)
Comment (aarsilv) [reply eppo-sdk-test/helpers/AssignmentLogDataTest.cs:31]: may be _too_fragile. Is 1 ms a healthy amount of buffer? should it be 5? 20?

**Review by typotter — APPROVED** [review]
Thanks for the quick fix!

**Pull request merged**